### ISSUE TITLE
graphviz: same map id is assigned for different graphs

### ISSUE THREAD

**Issue by tk0miya** [sphinx-doc/sphinx#4514]
### Problem
- When the document has multiple graphviz graphs
- same map id is assigned for them

#### Procedure to reproduce the problem
```
.. graphviz::

   digraph {
     A [href="URL"];
   }

.. graphviz::

   digraph {
     B [href="URL"];
   }
```

#### Error logs / results
Following HTML is generated. There are multiple `<map>` tags having `id="#%3"`. `usemap` attribute is also.
```
<img src="_images/graphviz-1d4985d223da0b0b87d99a9a48f61a022e1283be.png" alt="digraph {
  A [href=&quot;URL&quot;];
}" usemap="#%3" />
<map id="%3" name="%3">
<area shape="poly" id="node1" href="URL" title="A" alt="" coords="77,29,76,22,70,15,62,10,52,7,41,5,30,7,20,10,12,15,7,22,5,29,7,37,12
,43,20,49,30,52,41,53,52,52,62,49,70,43,76,37"/>
</map>
<img src="_images/graphviz-573176d0e453cc933d1043e425714fda81b96309.png" alt="digraph {
  B [href=&quot;URL&quot;];
}" usemap="#%3" />
<map id="%3" name="%3">
<area shape="poly" id="node1" href="URL" title="B" alt="" coords="77,29,76,22,70,15,62,10,52,7,41,5,30,7,20,10,12,15,7,22,5,29,7,37,12,43,20,49,30,52,41,53,52,52,62,49,70,43,76,37"/>
</map>
```

#### Expected results
- Generate valid HTML and correct clickable maps

### Environment info
- OS: Mac
- Python version: 2.7.14
- Sphinx version: 1.6.6

**Comment by tk0miya** [comment]
Note:
`%3` is comes from graphviz (see https://gitlab.com/graphviz/graphviz/issues/1327).
Anyway Sphinx should assign better IDs to every graphs.

**Comment by tk0miya** [comment]
Fixed by #4515
